### PR TITLE
Follow-up cleaning & fixes for compatibility with NumPy 1 & 2

### PR DESCRIPTION
**Pull request by lagru**
## Description

A follow-up to #7288 addressing a few review comments that were left pending.

## Checklist

- A descriptive but concise pull request title
- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [Unit tests](https://scikit-image.org/docs/dev/development/contribute.html#testing)
- A gallery example in `./doc/examples` for new features
- [Contribution guide](https://scikit-image.org/docs/dev/development/contribute.html) is followed

## Release note

Summarize the introduced changes in the code block below in one or a few sentences. The
summary will be included in the next release notes automatically:

```release-note
...
```

### CODE REVIEW

**Comment by lagru** [comment]
@stefanv, let me know if I shall revert the deprecation of `lookfor` here as well (see https://github.com/scikit-image/scikit-image/pull/7288#discussion_r1499643081).

**Comment by stefanv** [comment]
Is there a reason to deprecate `lookfor`? That code hasn't been touched in a while, and works fine. It's hard to know whether it is commonly used, but it is still part of our docs / tutorials. Perhaps we do a cleanup of all of those resources together?

**Comment by lagru** [comment]
> works fine

In https://github.com/scikit-image/scikit-image/pull/7288#discussion_r1455646928 it was mentioned that there might be some issues. But I agree that's probably not a reason enough to change the status quo. I'll revert the deprecation an we can deal with that down the road. :+1:

File: doc/source/user_guide/getting_help.rst [hunk 1]
```diff
@@ -26,6 +26,16 @@ Use the ``quick search`` field in the navigation bar of the online
...
 rescaling, denoising, etc.) in the documentation.
 
+
+API Discovery
+-------------
+
+We provide a ``lookfor`` function to search API functions::
+
+   import skimage as ski
+   ski.util.lookfor('eigenvalue')
```
Comment (lagru) [inline doc/source/user_guide/getting_help.rst:36]: Note the change from `ski.lookfor` to `ski.util.lookfor`. It's still available as the former, but not advertised. Now, `ski.util.lookfor` is also included in the HTML API doc.

**Review by stefanv — APPROVED** [review]

**Pull request merged**